xarray/core/extension_array.py

```diff
@@ -123,8 +123,6 @@ def __setitem__(self, key, val):
         self.array[key] = val
 
     def __eq__(self, other):
-        if np.isscalar(other):
-            other = type(self)(type(self.array)([other]))
```

Comment:
so we just defer to pandas to handle the scalar case?

Reply:
That would be the idea.